lib/mapUtils.js

```diff
@@ -43,3 +48,178 @@ export async function getProductData(store) {
+  const hoursPerDayDict = {};
+
+  if (hoursStrSplit.length === 2 && hoursStrSplit.includes('Daily')) {
+    const time = hoursStrSplit[0];
+
+    daysOfTheWeekFull.map((day, _) => {
+      hoursPerDayDict[day] = time;
+    });
+  }
+
+  // Case: There exist different hours on certain days
+  // e.g. 9am-8pm Tu-Sat, 9am-7pm Sun, Closed Mon
+  else if (hoursStrSplit.length > 2) {
+    // e.g. timeAndDayPairs = ['9am-8pm Tu-Sat', '9am-7pm Sun', ...]
+    const timeAndDayPairs = formattedHours.split(', ');
+
+    timeAndDayPairs.map((timeAndDayPair, _) => {
+      // e.g. timeAndDaySplit = ['9am-8pm', 'Tu-Sat']
+      const timeAndDaySplit = timeAndDayPair.split(' ');
+      // e.g. 9am-8pm
+      const time = timeAndDaySplit[0];
+      // e.g. [Tu, Sat] or [Mon]
+      const days = timeAndDaySplit[1].split('-');
+
+      // Case: Time corresponds to one day
+      if (days.length === 1) {
+        hoursPerDayDict[daysOfTheWeek[days[0]]] = time;
+      } else if (days.length === 2) {
+        const firstDay = days[0].charAt(0).toUpperCase() + days[0].slice(1);
+        const lastDay = days[1].charAt(0).toUpperCase() + days[1].slice(1);
+
+        // ['Sun', 'Mon', 'Tues', ...]
+        const indexFirstDay = daysOfTheWeekAbbrev.indexOf(firstDay);
+        const indexLastDay = daysOfTheWeekAbbrev.indexOf(lastDay);
+
+        let numDaysInInterval;
+
+        if (indexLastDay > indexFirstDay) {
+          // Case: Interval is Sun-Thurs, 0-4
+          numDaysInInterval = indexLastDay - indexFirstDay + 1;
+        } else {
+          // Case: Interval is Thurs-Sun, 4-0
+          numDaysInInterval = indexLastDay - indexFirstDay + 8;
+        }
+
+        [...Array(numDaysInInterval).keys()].map((i, _) => {
+          const dayIndex = (indexFirstDay + i) % 7;
+          hoursPerDayDict[daysOfTheWeekFull[dayIndex]] = time;
+        });
+      } else {
+        // Shouldn't ever have a case with more than two days in an interval.
+        console.log(
+          '[mapUtils] constructHoursDict: Incorrect store hours formatting in Airtable.'
+        );
+        return null;
+      }
+    });
+  }
+  // If the hours for that day are not available, set it as 'Closed'.
+  daysOfTheWeekFull.map(day => {
+    if (!Object.keys(hoursPerDayDict).includes(day)) {
+      hoursPerDayDict[day] = 'Closed';
+    }
+  });
+  return hoursPerDayDict;
+}
+
+// Takes in the index representing today's day of the week
+// and a dictionary mapping days to store hours. Returns a string
+// indicating when the store will next be open.
+function computeNextOpenDay(todaysDayIndex, hoursPerDayDict) {
+  // The store is closed, so we need the next day and its opening time
+  let tmrwDayIndex = (todaysDayIndex + 1) % 7;
+  let tmrwDay = daysOfTheWeekFull[tmrwDayIndex];
+  let tmrwStoreHours = hoursPerDayDict[tmrwDay];
+  // Loop until next open day is found.
+  while (tmrwStoreHours === 'Closed') {
```

Comment:
NICE that you catch this! I was expecting this edge case `9am-7pm Sun, 9am-8pm Tu-Thurs, Closed Fri, Closed Sat` to break but that's very cool that it works! Still, I'd probably rename this to be more specific like next opening day bc 'tmrw' is a bit deceiving.